### DISCUSSION TITLE
Stokes drift

### DISCUSSION THREAD

**Discussion by larsarneborg** [OpenDrift/opendrift#1062] [Q&A]
We are using OpenDrift in an archipelago area and would like to test the influence of Stokes drift without setting up a wave model. Is it possible to get OpenDrift to calculate fetch for each grid point and wind direction, calculate wave parameters (simplified), and Stokes drift? Or in other words, what is the minimum amount of information we need to supply to include Stokes drift in the calculations? I don't find much information on the Stokes drift option.

**Comment by knutfrode** [comment]
Hi,

You are right, the documentation is lacking for the options on Stokes drift, this should be improved.

If Stokes drift is available from a wave model (preferred option), limited fetch is already handled properly, but this is not always readily available.

If Stokes drift is not available, presently the only option is to use a constant fraction of the wind, by increasing the `wind_drift_factor` from e.g. 2% (surface wind drag but no Stokes) to 3.5%, where a Stokes drift of 1.5% of the wind speed is added, and assumed to be in the same direction.
In this case there is no compensation for limited fetch, except that there is a possibility to reduce the given Stokes fraction (e.g. `wind_drift_factor` of 2.5%). However, this reduction is then the same for the whole simulation domain/period and wind/wave directions (though it can be different for each particle).
Thus, presently the minimum information needed to include Stokes drift (if not explicitly available from wave model), is to adjust the `wind_drift_factor`, and to have winds available.

In fact, we have been looking very recently on how to include a more proper handling of limited fetch. Thus we plan to fairly soon (~few months) add options to calculate the fetch for all positions in a simulation domain for tabulated wind/wave directions, either dynamically at the start of the simulation, of by pre-generating this before the run.
We are open to suggestions on how this best could be done.

**Reply by larsarneborg** [reply]
Thanks!
